wordpress-mobile/WordPress-Android
# Issue/13328 restore details

Parent #13328

This PR implements the "details" view (a.k.a. the select options) for the restore request.

Although RestoreDetailsViewModel was cloned off of BackupDownloadDetailsViewModel, the logic deviates because of the introduction of the Warning step in restore. 

**Cloned Classes**
RestoreListItemState
RestoreDetailsAdapter
RestoreDetailsFragment
RestoreSubHeaderViewHolder
RestoreDetailsStateListItemBuilder
restore_list_subheader_item.xml 

Notes:
- Ignore styling - this is being handled separately 
- Ignore unit test placement in packages - this is being handled separately 
- RestoreActivity & RestoreViewModel (+unit tests) will continue to get updated as I move throughout the steps of the wizard

To test:
- Launch the app with RestoreFeatureConfig flag on and the BackupFeatureFlag on
- Navigate to ActivityLog
- Tap the more menu
- Select the restore option
- Note (1) that the detail view is shown (2) the checkboxes can be selected/unselected, tapping on X or back button closes the view

PR submission checklist:
